.github/workflows/lint.yaml

```diff
@@ -0,0 +1,20 @@
+name: Ruff Linting
```

Comment:
Does this make sense as a full separate action or as parallelized steps within a consolidated "check" action?

Reply:
yea, either one, I usually just do separate files as a lazy man's parallelization, but doing it explicitly in one is fine with me too, your call